apps/web/src/services/dashboards/widgets/total-failure-and-severity/TotalFailureAndSeverityWidget.vue

```diff
@@ -0,0 +1,315 @@
+});
+const state = reactive({
+    ...toRefs(useWidgetState(props)),
+    chartData: computed<XYChartData[]>(() => state.data),
+    dateRange: computed<DateRange>(() => {
+        const end = dayjs.utc(state.settings?.date_range?.end).format(DATE_FORMAT);
+        const start = dayjs.utc(end).subtract(11, 'month').format(DATE_FORMAT);
+        return { start, end };
+    }),
+    severityData: computed(() => {
+        const results: any[] = []; // TODO: fix type after get real data
+        Object.values(SEVERITY_STATUS_MAP).forEach((severity) => {
+            if (severity.name === 'pass') return;
+            const currValue = random(0, 200);
+            const prevValue = random(0, 200);
+            results.push({
+                ...severity,
+                value: currValue,
+                diff: currValue - prevValue,
+                rgb: getRGBFromHex(severity.color),
+            });
+        });
+        return results;
+    }),
+});
+const widgetFrameProps:ComputedRef = useWidgetFrameProps(props, state);
+
+/* API */
+const fetchData = async (): Promise<any> => {
+    try {
+        // TODO: change to real data
+        const results = await new Promise<any>((resolve) => {
+            setTimeout(() => {
+                resolve([
+                    { date: '2022-06', value: random(500, 1200) },
+                    { date: '2022-07', value: random(500, 1200) },
+                    { date: '2022-08', value: random(500, 1200) },
+                    { date: '2022-09', value: random(500, 1200) },
+                    { date: '2022-10', value: random(500, 1200) },
+                    { date: '2022-11', value: random(500, 1200) },
+                    { date: '2022-12', value: random(500, 1200) },
+                    { date: '2023-01', value: random(500, 1200) },
+                    { date: '2023-02', value: random(500, 1200) },
+                    { date: '2023-03', value: random(500, 1200) },
+                    { date: '2023-04', value: random(500, 1200) },
+                    { date: '2023-05', value: random(500, 1200) },
+                ]);
+            }, 2000);
+        });
+        return results;
+    } catch (e) {
+        ErrorHandler.handleError(e);
+        return [];
+    }
+};
+
+/* Util */
+const drawChart = (chartData: XYChartData[]) => {
+    const { chart, xAxis } = chartHelper.createXYDateChart({}, getDateAxisSettings(state.dateRange));
+    xAxis.get('baseInterval').timeUnit = 'month';
+    chartHelper.setChartColors(chart, colorSet.value);
+
+    const seriesSettings = {
+        name: 'value',
+        valueYField: 'value',
+    };
+    const series = chartHelper.createXYLineSeries(chart, seriesSettings);
+    chart.series.push(series);
+    // set data processor
+    series.data.processor = chartHelper.createDataProcessor({
+        dateFormat: DATE_FORMAT,
+        dateFields: [DATE_FIELD_NAME],
+    });
+    series.data.setAll(cloneDeep(chartData));
+};
+
+const initWidget = async () => {
```

Comment:
suggestion: I suggest to apply type declaration.

```suggestion
const initWidget = async (data?: Data): Promise<Data> => {
```